Bump to PostgREST v9 (#468)
PostgREST/postgrest-docs: releases/v9.0.0.rst

PostgREST version 9.0
=====================

> This mayor version is released following the PostgreSQL 14 launch and is accompanied with new features, improvements to functionality and bug fixes. Download the new version in the `PostgREST v9.0.0 release page <https://github.com/PostgREST/postgrest/releases>`_.

Reviewer: ```suggestion
This major version is released following the PostgreSQL 14 launch and is accompanied with new features, improvements to functionality and bug fixes. Download the new version in the `PostgREST v9.0.0 release page <https://github.com/PostgREST/postgrest/releases>`_.
```

Since dashes cannot be used on PostgreSQL 14, the GUC names are changed to the ones :ref:`mentioned in this section <guc_req_headers_cookies_claims>`. On older PostgreSQL versions we still use the :ref:`guc_legacy_names`. If you wish to use the new JSON GUCs on these versions, set the :ref:`db-use-legacy-gucs` config option to false.

Embedding on Partitioned Tables (and other features)
----------------------------------------------------

You can now confidently :ref:`embed partitioned tables <embedding_partitioned_tables>` with other tables; meanwhile, their partitions will stay hidden and won't interfere in the process. Likewise, other supported features for partitioned tables are: UPSERT, INSERT with Location response, OPTIONS request and OpenAPI support.

Dropping support for PostgreSQL 9.5
-----------------------------------

> Due to the recent `end of life of PostgreSQL 9.6 <https://www.postgresql.org/support/versioning/>`_, we are dropping support for the next older version. Take into consideration that we are also planning on dropping the support for 9.6 in the next minor versions.

Reviewer: ```suggestion
Due to the recent `end of life of PostgreSQL 9.6 <https://www.postgresql.org/support/versioning/>`_, we are dropping support for the next older version. Take into consideration that we are also planning on dropping the support for 9.6 in the next major version.
```

I don't think we can drop support in a minor version. That would be a breaking change.

Other features
--------------

* Add ``Retry-After`` header when recovering the connection. See :ref:`automatic_recovery`.

* Allow calling a function with a :ref:`single unnamed parameter <s_proc_single_unnamed>` to POST raw ``json/jsonb``, ``bytea`` or ``text``.

* Allow escaping inside double quotes with a backslash, e.g. ``?col=in.("Double\"Quote")``, ``?col=in.("Back\\slash")``. See :ref:`reserved-chars`.

* Allow filtering top-level resource based on embedded resources filters. This is enabled by adding ``!inner`` to the embedded resource. See :ref:`embedding_top_level_filter`.

* Allow specifying ``unknown`` for the ``is`` :ref:`operator <operators>`.

* Improve error message for ambiguous embedding and add a relevant hint that includes unambiguous embedding suggestions.

Bug fixes
---------

> * Fix OpenAPI missing default values for String types and identify Array types as "array" instead of "string"

Reviewer: ```suggestion
* Fix OpenAPI missing default values for String types and identify Array types as "array" instead of "string".
```

* Fix RPC return type handling for RETURNS TABLE with a single column (regression of `#1615 <https://github.com/PostgREST/postgrest/pull/1615>`_).

> * Fix using single double quotes (``"``) and backslashes (``/``) as values on the "in" operator

Reviewer: ```suggestion
* Fix using single double quotes (``"``) and backslashes (``/``) as values on the "in" operator.
```

> * Fix schema cache query failing with standard_conforming_strings = off

Reviewer: ```suggestion
* Fix schema cache query failing with standard_conforming_strings = off.
```
